tokio/src/net/windows/named_pipe.rs

```diff
@@ -0,0 +1,173 @@
+//! Windows named pipes.
+
+use mio::windows::NamedPipe as MioNamedPipe;
+use winapi::um::winbase::FILE_FLAG_OVERLAPPED;
+
+use std::{
+    ffi::OsStr,
+    fs::OpenOptions,
+    io::{ErrorKind, IoSlice, Result},
+    os::windows::prelude::*,
+    pin::Pin,
+    task::{Context, Poll},
+};
+
+use crate::io::{AsyncRead, AsyncWrite, Interest, PollEvented, ReadBuf, Ready};
+
+/// Non-blocking windows named pipe.
+#[derive(Debug)]
+pub struct NamedPipe {
+    io: PollEvented<MioNamedPipe>,
+}
+
+impl NamedPipe {
+    fn from_mio(pipe: MioNamedPipe) -> Result<Self> {
+        let io = PollEvented::new(pipe)?;
+        Ok(Self { io })
+    }
+
+    /// Binds new pipe server to the given pipe name.
+    ///
+    /// # Errors
+    ///
+    /// I'll error if there exists another instance of a server bound to that name.
+    fn bind<A: AsRef<OsStr>>(addr: A) -> Result<Self> {
+        let pipe = MioNamedPipe::new(addr)?;
+        Self::from_mio(pipe)
+    }
+
+    /// Opens given pipe as a client.
+    ///
+    /// # Errors
+    ///
+    /// I'll error if there is no such pipe or if there is no servers available.
+    fn open<A: AsRef<OsStr>>(addr: A) -> Result<Self> {
+        let file = OpenOptions::new()
+            .read(true)
+            .write(true)
+            .custom_flags(FILE_FLAG_OVERLAPPED)
```

Comment:
Should we be setting [`security_qos_flags`](https://doc.rust-lang.org/std/os/windows/fs/trait.OpenOptionsExt.html#tymethod.security_qos_flags) here?